libvast/src/command.cpp

```diff
@@ -103,17 +104,18 @@ caf::error parse(command::invocation& result, const command& cmd,
   return parse(result, **i, position + 1, last);
 }
 
-caf::expected<command::invocation> parse(const command& root,
-                                         command::argument_iterator first,
-                                         command::argument_iterator last) {
+command::invocation parse(const command& root, command::argument_iterator first,
+                          command::argument_iterator last) {
   command::invocation result;
   if (auto err = parse(result, root, first, last))
-    return err;
+    result.error = std::move(err);
   return result;
 }
 
 caf::message run(command::invocation& invocation, caf::actor_system& sys) {
   auto& cmd = *invocation.target;
+  if (!invocation)
```

Comment:
Shoudln't this be checked before reaching into `*invocation`?

Reply:
Yep, good catch. Makes more sense to check the error first.